src/test/kotlin/com/pubnub/api/subscribe/eventengine/effect/HandshakeEffectTest.kt

```diff
@@ -0,0 +1,86 @@
+package com.pubnub.api.subscribe.eventengine.effect
+
+import com.pubnub.api.PubNubException
+import com.pubnub.api.endpoints.remoteaction.RemoteAction
+import com.pubnub.api.subscribe.eventengine.event.Event
+import com.pubnub.api.subscribe.eventengine.event.SubscriptionCursor
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+
+class HandshakeEffectTest {
+    private val channels = listOf("channel1")
+    private val channelGroups = listOf("channelGroup1")
+    private val handshakeProvider = mockk<HandshakeProvider>()
+    private val eventDeliver = SubscribeManagedEffectFactoryTest.TestEventDeliver()
+    private val handshakeInvocation = SubscribeEffectInvocation.Handshake(channels, channelGroups)
+    private val subscriptionCursor = SubscriptionCursor(1337L, "1337")
+    private val reason = PubNubException("Unknown error")
+
+    @Test
+    fun `should deliver HandshakeSuccess event when HandshakeEffect succeeded `() {
+        // given
+        every { handshakeProvider.handshake(any(), any()) } returns successfulRemoteAction(subscriptionCursor)
+        val handshakeEffect = HandshakeEffect(handshakeProvider, eventDeliver, handshakeInvocation)
+
+        // when
+        handshakeEffect.runEffect()
+
+        // then
+        Thread.sleep(50)
```

Comment:
I don't think we need any of those `Thread.sleep`. There's got to be a better, more reliant and faster way to do the same.

Reply:
Yes. I can fix that.